File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-2.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        left=1

        for right in range(1,len(nums)):

            if nums[right] != nums[right-1]:
                nums[left] = nums[right]
                left += 1
        return left

        # i = 0

        # while i < len(nums):
        #     if nums[i] in nums[:i]:
        #         nums.pop(i)
        #     else:
        #         i+=1
        # return len(nums)
        
        # l=1 

        # for r in range(1,len(nums)):
        #     if nums[r]!=nums[r-1]:
        #         nums[l]=nums[r]
        #         l+=1
        # return l

        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.rs
impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut left = 0;
        let mut right = numbers.len() - 1;


        while left < right {
            let current_sum = numbers[right]+numbers[left];

            if current_sum == target{
                return vec![left as i32 + 1,right as i32 + 1]

            } else if current_sum > target {
                right -= 1;

            } else  {
                left += 1;
        };
    };
    vec![]
}
}
